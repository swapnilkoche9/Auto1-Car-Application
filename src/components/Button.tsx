import React from 'react'
import {ButtonProps} from '../utils/interface'

const Button = (props:ButtonProps) => (
  <button className='btn' onClick={props.handleClick}>{props.text}</button>
)

export default Button
