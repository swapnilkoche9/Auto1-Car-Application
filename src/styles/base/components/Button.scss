.btn {
  background-color: $orange;
  color: $pure-white;
  padding: $xs-size;
  margin: $m-size;
  width: $btnWidth-size;
  height: $btnHeight-size;
  border: none;
  border-radius: $xxs-size;
  cursor: pointer;
  outline: none;
  font-family: 'Roboto', sans-serif;
  font-weight: 400;
  font-size: $m-size
}

.btn:hover {
  background-color: $dark-orange;
  color: $light-gray;
}
.btn:focus {
  background-color: $dark-orange;
  color: $light-gray;
}
