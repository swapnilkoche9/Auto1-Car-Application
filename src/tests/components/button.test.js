import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Button from '../../components/Button'

Enzyme.configure({ adapter: new Adapter() })
const { shallow } = Enzyme

const handleClick = jest.fn()
const wrapper = shallow(
  <Button
    handleClick={handleClick}
    text={'Save'}
  />
)
test('should render Button correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should render Button with alt data correctly', () => {
  wrapper.setProps({
    text: 'Remove'
  })
  expect(wrapper).toMatchSnapshot()
})
