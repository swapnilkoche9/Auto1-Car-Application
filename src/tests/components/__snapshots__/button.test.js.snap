// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Button correctly 1`] = `ShallowWrapper {}`;

exports[`should render Button with alt data correctly 1`] = `ShallowWrapper {}`;
